pkg/provider/aws/secretsmanager/secretsmanager.go

```diff
@@ -237,15 +250,31 @@ func (sm *SecretsManager) PushSecret(ctx context.Context, value []byte, _ *apiex
 	if awsSecret != nil && bytes.Equal(awsSecret.SecretBinary, value) {
 		return nil
 	}
+
 	input := &awssm.PutSecretValueInput{
-		SecretId:     awsSecret.ARN,
-		SecretBinary: value,
+		SecretId:           awsSecret.ARN,
+		SecretBinary:       value,
+		ClientRequestToken: bumpVersionNumber(awsSecret.VersionId),
 	}
 	_, err = sm.client.PutSecretValueWithContext(ctx, input)
 	metrics.ObserveAPICall(constants.ProviderAWSSM, constants.CallAWSSMPutSecretValue, err)
 	return err
 }
 
+func bumpVersionNumber(id *string) *string {
+	if id == nil {
+		output := fmt.Sprintf("%032d", 1)
+		return &output
+	}
+	oldVersion, err := strconv.ParseInt(*id, 10, 64)
+	if err != nil {
+		output := fmt.Sprintf("%032d", 1)
+		return &output
```

Comment:
If there's a integer overflow it will error, we can not create another version `00000000000000000000000000000001`. This could happen if someone modifies the secret manually (which sets a uuid which is not in range of a 64bit int).

theoretically someone could create a version manually through CLI with `00000000000018446744073709551615` (uint64 max) and then ESO tries to push a version `0001` which will fail. I think that both aws cli and console will generate a uuid which always will be in a range > uint64, hence it will always break eso after manually modifying the secret.

Not sure if we should convert to a bigint type to decrease the chances of this happening. Anyways it would be good to have good user feedback on this. Maybe returning an error here would be a good option. WDYT?

Reply:
@moolen FYI: I'm currently trekking in the mountains and only have my phone with me, I'll be back next week

Reply:
Enjoy your time :mountain: :hiking_boot: :sun_with_face:

Reply:
>Secrets Manager removes deprecated versions when there are more than 100. Secrets Manager doesn't remove versions created less than 24 hours ago.

Didn't realise SM kept up to 100 old versions. 
I've switched both to using a bigint and using an incrementing uuid as the version number.
Overflow can still happen when the previous version was `FFFFFFFF-FFFF-FFFF-FFFF-FFFFFFFFFFFF`: The next version would be `00000000-0000-0000-0000-000000000000`